### PR TITLE
WT-7721 Update test-format to reopen an existing database with different config

### PR DESCRIPTION
**Pull request by raviprakashgiri29**
The test-format binary can now reopen an already existing database with different CONFIG files because some configurations are not supported by the older release. In compatibility testing, we want the test format binary to operate on the CONFIG that it understands.

### CODE REVIEW

**Review by tammybailey — APPROVED** [review]
lgtm

File: test/format/t.c [hunk 1]
```diff
@@ -218,8 +218,12 @@ main(int argc, char *argv[])
...
      */
     if (g.reopen) {
-        if (config != NULL)
-            testutil_die(EINVAL, "-c incompatible with -R");
+        /*
+         * The test-format binary can now reopen an already existing database with different CONFIG
+         * file because some configurations are not supported by the older release and in
+         * compatibility testing we want the test format binary to operate on the CONFIG that it
+         * understands.
+         */
```
Comment (lukech) [inline test/format/t.c:226]: Thanks for adding the descriptive comment. Can a blank line be added after this new comment section, so that it does not look to be related to the line (227) immediately after it?

**Review by lukech — APPROVED** [review]

File: test/format/t.c (hunk 1 above)
Comment (keithbostic) [reply test/format/t.c:226]: I'm not sure if this comment is a good idea.

What I'm trying to say is code doesn't generally have comments describing  "things we don't do", that's going to get messy if we keep adding comments describing things the code doesn't do.

**Review by keithbostic — APPROVED** [review]

File: test/format/t.c (hunk 1 above)
Comment (raviprakashgiri29) [reply test/format/t.c:226]: @keithbostic Yeah, I agree not to mention the things that we don't do. I will better add this comment in the PR descriptions.
